### ISSUE TITLE
Add support for webRTC

### ISSUE THREAD

**Issue by toots** [savonet/liquidsoap#60]
There seems to be a good lib for that there:
  https://code.google.com/p/webrtc

**Comment by Keyne** [comment]
How is this on nowadays? Is it possible to do it using gstreamer? (https://github.com/centricular/gstwebrtc-demos) Liquidsoap is already integrated with gstreamer.  And it's already possible do have a RTP input, although I've not find I way to create the answer for the webrtc offer after loading the SDP.

**Comment by smimram** [comment]
Yes, apparently support has recently been integrated in GStreamer :
http://blog.nirbheek.in/2018/02/gstreamer-webrtc.html

Let's reopen this bug :)

**Comment by romansavrulin** [comment]
Any news on this?

**Comment by smimram** [comment]
Nope. However, we're now moving toward ffmpeg. If you manage to get something working with ffmpeg command line, chances are high that we can have it quite quickly in Liquidsoap.

**Comment by romansavrulin** [comment]
Nice move! Will try to get it working!

**Comment by smimram** [comment]
Great, please keep us posted!

**Comment by toots** [comment]
Don't hesitate to reach out in our slack channel if you need!

**Comment by smimram** [comment]
Some notes for later
- [it seems that ffmpeg does not natively handles webRTC](https://stackoverflow.com/questions/59865405/use-webrtc-getusermedia-stream-as-input-for-ffmpeg)
- [however GStreamer does handle webRTC](https://gstreamer.freedesktop.org/documentation/webrtc/index.html?gi-language=c)

There are a few libraries implementing webRTC:
- [in C](https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/)
- [in go](https://github.com/pion/webrtc)
- [in python](https://github.com/aiortc/aiortc)

**Comment by danrossi** [comment]
I'll add to this open ticket. Can this support whepsrc / webRTC via gstreamer ? The playout is required to go to the audiosink on a PI for playout. So playback client rather than restream
